File: core/simulator/ArduinoState.cpp
//
// Created by TIZI on 08/12/2024.
//

#include "ArduinoState.hpp"

namespace arguino::simulator {

bool ArduinoState::get_digital(pin_t pin) const
{
    return _digitalPins[pin];
}

bool ArduinoState::set_digital(pin_t pin, digital_t value)
{
    _digitalPins[pin] = value;
    return true;
}

PinMode ArduinoState::get_pin_mode(pin_t pin) const
{
    return _pinModes[pin];
}

bool ArduinoState::set_pin_mode(pin_t pin, PinMode mode)
{
    _pinModes[pin] = mode;
    return true;
}

}

File: gui/TcpAdapter/ArduinoState.cs
namespace TcpAdapter;

public struct ArduinoState {
    public const uint ANALOG_PIN_COUNT = 6;
    public const uint DIGITAL_PIN_COUNT = 14;

    public double[] AnalogPins { get; private set; } = new double[ANALOG_PIN_COUNT];
    public bool[] DigitalPins { get; private set; } = new bool[DIGITAL_PIN_COUNT];
    
    public enum PinMode { In, Out };
    public PinMode[] PinModes { get; private set; } = new PinMode[DIGITAL_PIN_COUNT];

    public ArduinoState() {}
}